src/Illuminate/Queue/Jobs/Job.php

```diff
@@ -80,7 +81,23 @@ public function fire()
 
         [$class, $method] = JobName::parse($payload['job']);
 
-        ($this->instance = $this->resolve($class))->{$method}($this, $payload['data']);
+        $this->instance = $this->resolve($class);
+
+        $handler = function () use ($method, $payload) {
+            $this->instance->{$method}($this, $payload['data']);
+        };
+
+        if ($this->instance instanceof Transactional) {
+            $handler = function () use ($handler) {
+                $connection = method_exists($this->instance, 'dbConnection') && \is_callable([$this->instance, 'dbConnection'])
```

Comment:
How can we, except with documentation, better indicate to developers that this `dbConnection` method is possible to use?

The interface doesn't indicate it.

Also I'd like to see the naming a bit improved. E.g. `getConnectionForTransaction` or something, otherwise it's ambiguous.

Reply:
https://github.com/laravel/framework/blob/5.7/src/Illuminate/Queue/Jobs/Job.php#L171
https://github.com/laravel/framework/blob/5.7/src/Illuminate/Bus/Dispatcher.php#L156
https://github.com/laravel/framework/blob/5.7/src/Illuminate/Queue/Queue.php#L151
https://github.com/laravel/framework/blob/5.7/src/Illuminate/Queue/Queue.php#L163
https://github.com/laravel/framework/blob/5.7/src/Illuminate/Broadcasting/BroadcastEvent.php#L43
https://github.com/laravel/framework/blob/5.7/src/Illuminate/Broadcasting/BroadcastEvent.php#L60
https://github.com/laravel/framework/blob/5.7/src/Illuminate/Broadcasting/BroadcastManager.php#L119
https://github.com/laravel/framework/blob/5.7/src/Illuminate/Foundation/Http/FormRequest.php#L150

What about interface for this and another ones?